### PR TITLE
Fixes #3712: Add accessibility support for ImageRegionSelectionInteraction [Removed Todo followed by #4497]

### LINKED ISSUE #3712: [A11Y] Add accessibility support for ImageRegionSelectionInteraction [Blocked: oppia/#9924]
ImageRegions of ImageRegionSelectionInteraction doesn't have accessibility support.

Additional information other than the exact answer need to be added as content description for each image region.

https://github.com/oppia/oppia/issues/9924 For details

### PR DESCRIPTION
**Pull request by vrajdesai78**
## Explanation
Fixes #3712: Removed TODO followed by PR #4497

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## Video

https://user-images.githubusercontent.com/43074241/183263993-d981cd7e-1a3d-4b1e-aaac-a9d08f6a8c06.mp4

### CODE REVIEW

File: instrumentation/src/javatests/org/oppia/android/instrumentation/player/ExplorationPlayerTest.kt
Comment (BenHenning) [inline instrumentation/src/javatests/org/oppia/android/instrumentation/player/ExplorationPlayerTest.kt:61]: I'm not sure simply removing the TODO is actually addressing this. It seems as though this intends to update the test to retrieve the region using its content description.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @vrajdesai78. I had one comment, but otherwise isn't this duplicating work already checked into develop from #4497? Can those deltas be removed since they're no longer needed?

**Comment by BenHenning** [comment]
FWIW I suspect that you need to merge in the latest develop to remove the duplicated deltas.

File: instrumentation/src/javatests/org/oppia/android/instrumentation/player/ExplorationPlayerTest.kt
Comment (vrajdesai78) [reply instrumentation/src/javatests/org/oppia/android/instrumentation/player/ExplorationPlayerTest.kt:61]: I haven't worked with e2e test, so I need to check how we can retrieve region using its content description. Also, for these test it is mentioned that it `Need backend support to test the ImageRegionSelectionInteraction`. So, is it mandatory to update this test as I think still we don't have backend support for ImageRegionSelectionInteraction

**Comment by vrajdesai78** [comment]
> Thanks @vrajdesai78. I had one comment, but otherwise isn't this duplicating work already checked into develop from #4497? Can those deltas be removed since they're no longer needed?

You are correct this changes are already merged with previous PR and for this PR I have already merged with latest develop still it shows changes. So, I think I need to create a new branch instead of using older branch to remove this duplicate deltas? WDYT @BenHenning

**Review by rt4914 — APPROVED** [review]
Approved for code-owners and based on previous PR

**Comment by BenHenning** [comment]
> > Thanks @vrajdesai78. I had one comment, but otherwise isn't this duplicating work already checked into develop from #4497? Can those deltas be removed since they're no longer needed?
> 
> You are correct this changes are already merged with previous PR and for this PR I have already merged with latest develop still it shows changes. So, I think I need to create a new branch instead of using older branch to remove this duplicate deltas? WDYT @BenHenning

@vrajdesai78 are you sure that your fork's develop is up-to-date with Oppia Android's develop branch? I just want to make sure since the situation seems a bit odd.

We shouldn't need to recreate the branch. If the above doesn't help, I suggest reaching out to me on chat and we can work through steps to try and address it.

**Comment by vrajdesai78** [comment]
> > > Thanks @vrajdesai78. I had one comment, but otherwise isn't this duplicating work already checked into develop from #4497? Can those deltas be removed since they're no longer needed?
> > 
> > 
> > You are correct this changes are already merged with previous PR and for this PR I have already merged with latest develop still it shows changes. So, I think I need to create a new branch instead of using older branch to remove this duplicate deltas? WDYT @BenHenning
> 
> @vrajdesai78 are you sure that your fork's develop is up-to-date with Oppia Android's develop branch? I just want to make sure since the situation seems a bit odd.
> 
> We shouldn't need to recreate the branch. If the above doesn't help, I suggest reaching out to me on chat and we can work through steps to try and address it.

I hope now develop branch is merged properly. @BenHenning

File: instrumentation/src/javatests/org/oppia/android/instrumentation/player/ExplorationPlayerTest.kt
Comment (BenHenning) [reply instrumentation/src/javatests/org/oppia/android/instrumentation/player/ExplorationPlayerTest.kt:61]: The backend support is still needed for all aspects of the new content description, it's just that we're pushing the requirement down to the import and future download pipelines (the latter of which hasn't been built yet, and will also break this test once it does).

Regarding how to do this, I think you can use https://developer.android.com/reference/androidx/test/uiautomator/UiSelector#description(java.lang.String) or other UiSelector methods in conjunction with UiDevice.findObject().

**Review by BenHenning — COMMENTED** [review]
Great to hear that the merging is addressed now @vrajdesai78. I've followed up to your other comment (apologies, I think I missed it earlier).

**Comment by vrajdesai78** [comment]
Keep this PR open

**Comment by vrajdesai78** [comment]
Keep this PR open

**Comment by vrajdesai78** [comment]
Keep this PR open

**Comment by vrajdesai78** [comment]
Keep this PR open

**Comment by vrajdesai78** [comment]
I have filed a new issue #4639 to fetch image region from it's content description in end-to-end tests. So, in this PR I have have updated issue number in ToDo. Can we merge this PR and close issue #3712 now as seperate issue for end-to-end tests is already filed.

File: instrumentation/src/javatests/org/oppia/android/instrumentation/player/ExplorationPlayerTest.kt
Comment (vrajdesai78) [reply instrumentation/src/javatests/org/oppia/android/instrumentation/player/ExplorationPlayerTest.kt:61]: Can we close this PR now as mentioned in this [comment](https://github.com/oppia/oppia-android/pull/4543#issuecomment-1267318200)

**Comment by BenHenning** [comment]
> I have filed a new issue #4639 to fetch image region from it's content description in end-to-end tests. So, in this PR I have have updated issue number in ToDo. Can we merge this PR and close issue #3712 now as seperate issue for end-to-end tests is already filed.

@vrajdesai78 please add a video to this PR manually demonstrating what the end-to-end test would be verifying. That part is important before we consider #3712 is done without updating the end-to-end tests.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @vrajdesai78. PTAL at my latest comment. That needs to be addressed before I can approve & merge this.

**Comment by vrajdesai78** [comment]
> > I have filed a new issue #4639 to fetch image region from it's content description in end-to-end tests. So, in this PR I have have updated issue number in ToDo. Can we merge this PR and close issue #3712 now as seperate issue for end-to-end tests is already filed.
> 
> @vrajdesai78 please add a video to this PR manually demonstrating what the end-to-end test would be verifying. That part is important before we consider #3712 is done without updating the end-to-end tests.

Thanks @BenHenning, I have added video in this PR description as well as in #4639 also.

**Review by BenHenning — APPROVED** [review]
Thanks @vrajdesai78. LGTM.

Also for posterity, we're fine with removing this TODO because end-to-end tests aren't quite 100% stable/ready to run, and getting them working has taken longer than expected. We'll need to update & verify the test passes.

**Pull request merged**
